pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -358,9 +369,21 @@ message LinuxUser {
     repeated int64 additional_gids = 3;
 }
 
-message ContainerConfig {
-    // Name of the container.
+// ContainerMetadata holds all necessary information for building unique
+// container name.
+message ContainerMetadata {
+    // The name of the container. Same as the container name in the PodSpec.
     optional string name = 1;
+    // The restart count of the container.
+    optional uint32 restart_count = 2;
```

Comment:
I think it has to be monotonic for safety. If the kubelet sees state A, calls create, crashes, and sees state A again, it should make the same call again so that the runtime acts in a consistent way based on kubelet's view of the world.